enhancements/node-tuning/containerize-tuned.md

```diff
@@ -0,0 +1,255 @@
+This will have to be implemented on RHEL + RHCOS - [RHEL-23928](https://issues.redhat.com/browse/RHEL-23928)
+
+### Workflow Description
+
+N/A
+
+### API Extensions
+
+No API changes
+
+### Topology Considerations
+
+#### Hypershift / Hosted Control Planes
+
+TBD
+
+#### Standalone Clusters
+
+#### Single-node Deployments or MicroShift
+
+How does this proposal affect the resource consumption of a
+single-node OpenShift deployment (SNO), CPU and memory?
+
+TBD
+
+### Implementation Details/Notes/Constraints
+
+* “Accumulating” NTO images during upgrades.
+* Even with “podmanized” TuneD, we’ll likely not be able to resolve issues such as [these](https://github.com/openshift/machine-config-operator/pull/2944#issuecomment-1029739561) and centralized the tuning into NTO – chicken-and-egg with default openshift TuneD profiles. Rendering them via MCO/installer, doing some built-in NTO bootstrap or using host-native TuneD would help.
+* We need to ensure TuneD is not blocking Kubelet system service while starting before it. 15s of TuneD starting would be an issue. 
+
+### Risks and Mitigations
+
+* The [pod label based matching](https://docs.openshift.com/container-platform/4.15/scalability_and_performance/using-node-tuning-operator.html#accessing-an-example-node-tuning-operator-specification_node-tuning-operator) must stay working as it was never officially deprecated although the official warning exists.
+  * Disabling that functionality when PerformanceProfile is in place should be considered as a viable option.
+
+* Extra reboot during can be caused from the machine config for installation/upgrade.
+  * This is potentially solvable by providing a MC manifest for the TuneD systemd service at bootstrap.
+
+* Telco Day-0:
+  * Stock tuned profiles and all the profiles provided via Tuned.spec.profile are kept in an in memory overlay of the daemonset container.
+  We might need to store the profiles files on the node via MachineConfig to support Day 0 tuning (= there no NTO daemon during bootstrap).
+
+
+### Drawbacks
+
+The implementation is invasive and refactors a prominent part of NTO in the way it used to apply tunings via the tuned daemonset. 
+It will not be possible to downport such a solution and the reported issues mentioned at the start will have to be resolved in another manner for older versions.
+
+## Open Questions [optional]
+
+ > 1. The new limitation is that every change to Tuned profile (even sysctls) will cause a reboot. Avoiding that for specific cases is possible (when no PerfProfile in use?) via updating the files through the daemonset, but that conflicts with the MCP based approach.
+ > 2. If [RHEL-23928](https://issues.redhat.com/browse/RHEL-23928) is not implemented , how do we keep files open, for example for the rtentsk plugin. Something new or tuned as a daemon?
```

Comment:
I think you would also require https://issues.redhat.com/browse/RHEL-23926 to be implemented. Initially, wouldn't we want to keep running tuned in daemon mode from the new systemd service? Then when both of these enhancements are complete we could choose to just run it, apply the config and exit?

Reply:
I believe Bart is correct.

> how do we keep files open, for example for the rtentsk plugin

There seems to be a confusion whether we'll be running TuneD in one-shot more or as daemon.  The current PoC still runs TuneD as a daemon, so you have rtentsk plug-in.

Reply:
Moving these points to Implementation Details/Notes/Constraints  section (notes mostly)